File: app/src/main/java/com/linderaredux/db/DataManager.java
package com.linderaredux.db;

import com.linderaredux.api.response.patient.Patient;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Observable;

public class DataManager {

    AppDbHelper appDbHelper;

    public DataManager(AppDbHelper appDbHelper) {
        this.appDbHelper = appDbHelper;
    }

    public Observable<List<Patient>> getAllPatients() {
        return appDbHelper.getAllPatients();
    }

    public Observable<Boolean> savePatientList(List<Patient> patientList) {
        return appDbHelper.savePatientList(patientList);
    }
}

File: app/src/main/java/com/linderaredux/db/DataManager.kt
package com.linderaredux.db

import com.linderaredux.api.response.patient.Patient

import javax.inject.Inject
import javax.inject.Singleton

import io.reactivex.Observable

class DataManager(internal var appDbHelper: AppDbHelper) {

    val allPatients: Observable<List<Patient>>
        get() = appDbHelper.allPatients

    fun savePatientList(patientList: List<Patient>): Observable<Boolean> {
        return appDbHelper.savePatientList(patientList)
    }
}
